Packet++/src/IcmpV6Layer.cpp

```diff
@@ -0,0 +1,138 @@
+#define LOG_MODULE PacketLogModuleIcmpV6Layer
+
+#include <EndianPortable.h>
+#include <IPv4Layer.h>
+#include <IPv6Layer.h>
+#include <IcmpV6Layer.h>
+#include <NdpLayer.h>
+#include <PacketUtils.h>
+#include <memory>
+#include <sstream>
+
+namespace pcpp
+{
+IcmpV6Layer::IcmpV6Layer(ICMPv6MessageTypes type, uint8_t code) : Layer()
+{
+	m_DataLen = sizeof(icmpv6hdr);
+	m_Data = new uint8_t[m_DataLen];
+	memset(m_Data, 0, m_DataLen);
+	m_Protocol = ICMPv6;
+
+    icmpv6hdr *hdr = getIcmpv6Header();
+	hdr->type = type;
+	hdr->code = code;
+}
+
+void IcmpV6Layer::parseNextLayer()
+{
+	uint8_t *payload = m_Data + getHeaderLen();
+	size_t payloadLen = m_DataLen - getHeaderLen();
+
+	switch (getMessageType())
+	{
+	case ICMPv6_NEIGHBOR_SOLICITATION: {
+		m_NextLayer = new NDPNeighborSolicitationLayer(payload, payloadLen, this, m_Packet);
+		break;
+	}
+	case ICMPv6_NEIGHBOR_ADVERTISEMENT: {
+		m_NextLayer = new NDPNeighborAdvertisementLayer(payload, payloadLen, this, m_Packet);
+		break;
+	}
+	}
+}
+
+ICMPv6MessageTypes IcmpV6Layer::getMessageType() const
+{
+	return (ICMPv6MessageTypes)getIcmpv6Header()->type;
+}
+
+uint16_t IcmpV6Layer::getChecksum() const
+{
+	return be16toh(getIcmpv6Header()->checksum);
+}
+
+uint8_t IcmpV6Layer::getCode() const
+{
+	return getIcmpv6Header()->code;
+}
+
+void IcmpV6Layer::computeCalculateFields()
+{
+	calculateChecksum();
+}
+
+std::string IcmpV6Layer::toString() const
+{
+	std::string messageTypeAsString;
+	ICMPv6MessageTypes type = getMessageType();
+	switch (type)
+	{
+	case ICMPv6_NEIGHBOR_SOLICITATION:
+		messageTypeAsString = "Neighbor Solicitation";
+		break;
+	case ICMPv6_NEIGHBOR_ADVERTISEMENT:
+		messageTypeAsString = "Neighbor Advertisement";
+		break;
+	default:
+		messageTypeAsString = "Unknown";
+		break;
+	}
+
+	std::ostringstream typeStream;
+	typeStream << (int)getMessageType();

```

Comment:
You are getting the message type twice in this function:

`ICMPv6MessageTypes type = getMessageType();`

`typeStream << (int)getMessageType();`

I would propose to reduce it to one call.